npm_and_yarn/lib/dependabot/npm_and_yarn/file_updater/npm_lockfile_updater.rb

```diff
@@ -185,6 +185,13 @@ def handle_npm_updater_error(error, lockfile)
             raise_resolvability_error(error, lockfile)
           end
 
+          # Some private registries return a 403 when the user is readonly
+          if error.message.include?("403 Forbidden")
+            source = error.message.match(%r{https?://(?<source>[^/]+)/}).
+                     named_captures.fetch("source")
+            raise PrivateSourceAuthenticationFailure, source
```

Comment:
What if this is the central registry? And how come we're doing this in the file updater rather than the update checker?

Reply:
Ah yeah good point, will add a check to make sure it's not global - this seems to only happen for some sub-dependencies when doing a full resolution so pretty sure it won't raise in the update checker

Reply:
👍 - think we already have a helper method for checking.